### ISSUE TITLE
Volume: create new, attach existing, make from snapshot (seems not working)

### ISSUE THREAD

**Issue by styk-tv** [test-kitchen/kitchen-openstack#131]
Tried both:

```
 1. kitchen-openstack-2.1.1
 2. kitchen-openstack-2.2.0.pre.1
``` 

tried adding adding block_device_mappings: section to .kitchen.yml
under root node, driver node and driver_config node just to make sure

tried making brand new, from snapshot, attach existing with no success. tried based on example settings provided in README.me but also other combinations.

tried with device vdb and vda (since there is boot drive already)

Each time the driver doesn't display a single volume message (from source some messages should appear) when creating the instance (kitchen create) It does the instance, confirms winrm (windows) and exists. Obviously I don't mean windows mounting or anything like that. I mean from the perspective of openstack, I have not found a way to attach any form of volume during kitchen create making new instance.

```
block_device_mappings:
     make_volume: true
     snapshot_id: 2b449f7c-181d-4867-ba80-8843d6c0f56d
     device_name: vdb
     availability_zone: nova
     delete_on_termination: true
     creation_timeout: 120
```

also tried make_volume=false, volume_id=,volume_size= combinations

Any hints would be greatly appreciated. It seems block_device_mappings is not being hit at all.

P.

**Comment by jjasghar** [comment]
@styk-tv is this related? https://github.com/test-kitchen/kitchen-openstack/issues/130

**Comment by styk-tv** [comment]
#130 is about OpenStack and Vagrant-Openstack feature that is not in kitchen-openstack.
#131 (this issue) is about existing advertised kitchen-openstack feature that is not working (or so it seems) in versions specified.

**Comment by jjasghar** [comment]
Ah understood. I just thought it was curious that two issues came in about Volumes recently, when there hasn't been any since I took over maintainership.

This was a feature added before I took over the gem, and unfortunately I don't have Cinder on any of my clouds; so I have to spend the cycles to set this up. I'd do my best to get it over the holidays, but its unlikely I'll be able to look at this till mid January. Hopefully someone else in the community can help you.

**Comment by NaomiReeves** [comment]
Hey @styk-tv I had this same problem and I opened up the gem to check the syntax. I think the ReadMe example might have an error. Try dropping the "s" from block_device_mapping, changing  `block_device_mappings` to  `block_device_mapping` and see if that works for you.

**Comment by jjasghar** [comment]
Oh that's awesome if true. Thanks!

**Comment by jjasghar** [comment]
@NaomiReeves was this ever confirmed?

**Comment by jjasghar** [comment]
I guess also, @styk-tv was this verified?

**Comment by NaomiReeves** [comment]
I never heard from @styk-tv but I know it was the cause of my issue. Dropping the "s" from the block_device_mapping made it work for me. I'd hate to update the example without confirmation though.

**Comment by jjasghar** [comment]
:metal: thanks for that. Yeah there aren't many reports of people using the `block_device_mapping(s)` so I'd love to confirm this too.

welp: https://github.com/test-kitchen/kitchen-openstack/blob/master/lib/kitchen/driver/openstack.rb#L67 that's a typo. I'll fix it now.

**Comment by jjasghar** [comment]
Fixed via: 149a5d451360e35d057965ff688647c994379e68

**Comment by lgeinert** [comment]
Does anyone know if you can do two device mappings in one kitchen.yml?
